Gems/RemoteTools/Code/Source/Utilities/RemoteToolsJoinThread.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RemoteToolsJoinThread.h"
+
+#include <AzCore/Name/Name.h>
+#include <AzNetworking/Framework/INetworking.h>
+
+#include <RemoteToolsSystemComponent.h>
+#include <Source/AutoGen/RemoteTools.AutoPackets.h>
+
+namespace RemoteTools
+{
+    RemoteToolsJoinThread::RemoteToolsJoinThread(int updateRate, RemoteToolsSystemComponent* component)
+        : TimedThread("RemoteTools::RemoteToolsJoinThread", AZ::TimeMs(updateRate))
+    {
+        m_remoteToolsComponent = component;
+    }
+
+    RemoteToolsJoinThread::~RemoteToolsJoinThread()
+    {
+        Stop();
+        Join();
+    }
+
+    void RemoteToolsJoinThread::OnUpdate([[maybe_unused]] AZ::TimeMs updateRateMs)
+    {
+        bool isRequestingConnection = false;
+        auto& networkInterfaces = AZ::Interface<AzNetworking::INetworking>::Get()->GetNetworkInterfaces();
+        for (auto registryIt = m_remoteToolsComponent->m_entryRegistry.begin();
+            registryIt != m_remoteToolsComponent->m_entryRegistry.end();
+             ++registryIt)
```

Comment:
Repeat a range based for loop can be used here with a structured binding
```suggestion
        for (const auto& [crcId, toolsRegistryEntry] : m_remoteToolsComponent->m_entryRegistry)
```